cloud/pkg/cloudstream/tunnelserver.go

```diff
@@ -169,3 +176,25 @@ func (s *TunnelServer) Start() {
 		return
 	}
 }
+
+func (s *TunnelServer) updateNodeKubeletEndpoint(nodeName string) {
+	getNode, err := client.GetKubeClient().CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		klog.Warningf("node %s not found", nodeName)
+		return
+	}
+	if err != nil {
+		klog.Warningf("get node %s failed with error: %s", nodeName, err)
+		return
+	}
+
+	getNode.Status.DaemonEndpoints.KubeletEndpoint.Port = int32(s.tunnelPort)
+	node, err := client.GetKubeClient().CoreV1().Nodes().UpdateStatus(context.Background(), getNode, metav1.UpdateOptions{})
+	if err != nil {
+		klog.Errorf("update node KubeletEndpoint Port failed with error: %s, node: %v, tunnelPort:%v",
```

Comment:
We should have some try then exit directly if err occurs in this func. @chenchunxiu 
Ref:  https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L792

Reply:
Done